### PR TITLE
Make sure that ssh driver gets an ip address

### LINKED ISSUE #10244: How to add IP address for minikube
When i tart minikube i am facing the error saying that failed to start ssh bare metal machine and how to provide an IP into it.
All the software are upto date

`~$ minikube start --driver=ssh
😄  minikube v1.17.0 on Debian kali-rolling (vbox/amd64)
✨  Using the ssh driver based on existing profile

🧯  The requested memory allocation of 2200MiB does not leave room for system overhead (total system memory: 2996MiB). You may face stability issues.
💡  Suggestion: Start minikube with less memory allocated: 'minikube start --memory=2200mb'

👍  Starting control plane node minikube in cluster minikube
🤦  StartHost failed, but will try again: config: please provide an IP address
😿  Failed to start ssh bare metal machine. Running "minikube delete" may fix it: config: please provide an IP address

❌  Exiting due to GUEST_PROVISION: Failed to start host: config: please provide an IP address`

### PR DESCRIPTION
**Pull request by afbjorklund**
BEFORE

```
✨  Using the ssh driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🤦  StartHost failed, but will try again: config: please provide an IP address
😿  Failed to start ssh bare metal machine. Running "minikube delete" may fix it: config: please provide an IP address

❌  Exiting due to GUEST_PROVISION: Failed to start host: config: please provide an IP address

😿  If the above advice does not help, please let us know: 
👉  https://github.com/kubernetes/minikube/issues/new/choose

```

AFTER

```
✨  Using the ssh driver based on user configuration

❌  Exiting due to MK_USAGE: No IP address provided. Try specifying --ssh-ip-address, or see https://minikube.sigs.k8s.io/docs/drivers/ssh/

```

Closes #10244

### CODE REVIEW

**Review by medyagh — APPROVED** [review]

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -1097,6 +1097,13 @@ func validateFlags(cmd *cobra.Command, drvName string) {
 		}
 	}
 
+	if driver.IsSSH(drvName) {
+		sshIPAddress := viper.GetString(sshIPAddress)
+		if sshIPAddress == "" {
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go:1102]: how about adding a validate IP to be a valid ip,
so if user enters a wrong format for the IP we fail with exit.Usage ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/start.go (hunk 1 above)
Comment (afbjorklund) [reply cmd/minikube/cmd/start.go:1102]: I was going to validate the whole SSH connection later, but can check the IP address now

**Review by ilya-zuyev — APPROVED** [review]
lgtm

**Pull request merged**